### PR TITLE
Dockerfile, documentation

### PR DESCRIPTION
**Pull request by scottfrazer**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -1,2 +1,151 @@
...
+```
+
+### DELETE /thurloe/:user_id/:key
+
+```
+http --print=hbHB DELETE http://localhost:8000/thurloe/uid/k2
+curl -X DELETE http://localhost:8000/thurloe/uid/k
+```
+
+If the key is not found, a 404 is returned:
```
Comment (mcovarr) [inline README.md]: what do the other APIs do if keys or users aren't found?

File: src/main/scala/thurloe/Main.scala [hunk 2]
```diff
@@ -8,15 +8,15 @@ import spray.can.Http
...
 
   // create and start our service actor
   val service = system.actorOf(Props[ThurloeServiceActor], "demo-service")
 
   implicit val timeout = Timeout(5.seconds)
-  // start a new HTTP server on port 8080 with our service actor as the handler
-  IO(Http) ? Http.Bind(service, interface = "localhost", port = 8080)
+
+  // start a new HTTP server on port 8000 with our service actor as the handler
+  IO(Http) ? Http.Bind(service, interface = "0.0.0.0", port = 8000)
```
Comment (mcovarr) [inline src/main/scala/thurloe/Main.scala]: it would be nice to pull the port from config rather than hardcoding

Comment (geoffjentry) [inline src/main/scala/thurloe/Main.scala]: We should make the port configurable but that's out of scope for this PR. I'll add a ticket, it should be trivial

**Comment by mcovarr** [comment]
:+1: modulo nitpicks

File: README.md (hunk 1 above)
Comment (scottfrazer) [inline README.md]: I will find out and update the docs

File: src/main/scala/thurloe/Main.scala (hunk 2 above)
Comment (kshakir) [inline src/main/scala/thurloe/Main.scala]: I believe this isn't robust to failures, but :+1: for now.

https://groups.google.com/a/broadinstitute.org/forum/#!topic/dsde-engineering/xV96poHj5IA
broadinstitute/agora#1
broadinstitute/vault-common#7

Comment (scottfrazer) [inline src/main/scala/thurloe/Main.scala]: Good to know, let me add a comment with those links at least.

**Pull request merged**